src/core/amount.js

```diff
@@ -614,10 +586,12 @@ function(quality, counterCurrency, counterIssuer, opts) {
     quality as stored :  5 USD          /  3000000 drops
     inverted          :  3000000 drops  /          5 USD
   */
-  var adjusted = opts.inverse ? inverse(value) : value;
-  var nativeAdjusted = adjusted;
+  const valueStr = mantissa.toString() + 'e' + offset.toString();
+  let nativeAdjusted = this._is_native ?
+    new XRPValue(valueStr) : new IOUValue(valueStr);
```

Comment:
Hmmm, this is another quirk of the prior design: it doesn't really make sense to convert a quality to an amount because a quality is a ratio and an amount doesn't support ratio dimensions.

There are a couple options: throw an exception when from_quality's second parameter is XRP (this would be a breaking change). Or we could put in a hack that uses IOUValue.toFixed to cast to an XRPValue without violating the precision constraint. We should try to avoid breaking changes since this is a refactoring pull request though. Maybe later we can do the breaking change in a separate pull request. Even better would be to just delete Amount.from_quality and replace it with a function (not in amount.js) called `parseQuality` that returns an IOUValue.